mava/components/updating/parameter_server.py

```diff
@@ -144,6 +147,24 @@ def on_parameter_server_init_start(self, server: SystemParameterServer) -> None:
         # Interrupt the system in case all the executors failed
         server.store.parameters["num_executor_failed"] = 0
 
+        # Initiate best performance network values
+        server.store.parameters["best_checkpoint"] = {}
+        for metric in server.store.metrics_checkpoint:
+            server.store.parameters["best_checkpoint"][metric] = {}
+            for agent_net_key in networks.keys():
+                server.store.parameters["best_checkpoint"][metric][
```

Comment:
Can we add a loop here?

Reply:
Furthermore, did you check that the two network values can be different? It seems that this is assigned by reference which would allow the best_checkpoint values to be updated if the network values are updated. Or does this not happen?

Reply:
I will check that is a deep copy or not because from the unit tests it seems to works fine but I will re check it

Reply:
You were right it was reference

Reply:
Solved!

Reply:
Great :slightly_smiling_face: